### PR TITLE
Deck creation, deletion

### PR DESCRIPTION
**Pull request by NikasNikasNikas**
Added deck creation and deletion. Added a button to DeckMenu that allows to go to deck creation page.

### CODE REVIEW

File: MementoMori/MementoMori.Server/Service/DeckHelper.cs [hunk 1]
```diff
@@ -86,6 +85,31 @@ public void UpdateDeck(EditedDeckDTO editedDeckDTO, Guid requesterId)
...
+                Cards = null,
+                CardCount = 0,
+            };
+            _context.Decks.Add(newDeck);
+            _context.SaveChanges();
+        }
+        public void DeleteDeck(Guid deckId)
+        {
+            var deck = _context.Decks.Include(d => d.Cards).FirstOrDefault(d => d.Id == deckId);
+            _context.Decks.Remove(deck);
```
Comment (KarolisPetrovskis) [inline MementoMori/MementoMori.Server/Service/DeckHelper.cs]: Make sure `deck` is not null here

File: MementoMori/MementoMori.Server/Service/DeckHelper.cs [hunk 2]
```diff
@@ -86,6 +85,31 @@ public void UpdateDeck(EditedDeckDTO editedDeckDTO, Guid requesterId)
...
+                Title = deck.Title,
+                Description = deck.Title,
+                Tags = deck.Tags,
+                Rating = 0,
+                RatingCount = 0,
+                Modified = DateOnly.FromDateTime(DateTime.Now),
+                Cards = null,
+                CardCount = 0,
+            };
+            _context.Decks.Add(newDeck);
```
Comment (KarolisPetrovskis) [inline MementoMori/MementoMori.Server/Service/DeckHelper.cs]: Should probably set all properties of the new deck here instead calling `UpdateDeck` right afterwards.

File: MementoMori/MementoMori.Server/Controllers/DeckController.cs [hunk 3]
```diff
@@ -122,5 +124,31 @@ public IActionResult EditDeck(EditedDeckDTO editedDeckDTO)
             }
             return Ok();
         }
+        [HttpPost("createDeck")]
+        public IActionResult CreateDeck(EditedDeckDTO createDeckDTO)
+        {
+            Guid deckGuid = Guid.NewGuid();
```
Comment (KarolisPetrovskis) [inline MementoMori/MementoMori.Server/Controllers/DeckController.cs]: Imo `deckId` should be generated and assigned in `DeckHelper.CreateDeck()`

File: MementoMori/MementoMori.Server/Controllers/DeckController.cs [hunk 4]
```diff
@@ -122,5 +124,34 @@ public IActionResult EditDeck(EditedDeckDTO editedDeckDTO)
...
+            var requesterId = _authService.GetRequesterId(HttpContext);
+            if (requesterId == null)
+                return Unauthorized();
+            try
+            {
+                return Ok(_deckHelper.CreateDeck(createDeckDTO, (Guid)requesterId));
+            }
+            catch (UnauthorizedEditingException ex)
+            {
+                return Unauthorized();
```
Comment (KarolisPetrovskis) [inline MementoMori/MementoMori.Server/Controllers/DeckController.cs]: If I'm not mistaken this will never be triggered.

Comment (NikasNikasNikas) [reply MementoMori/MementoMori.Server/Controllers/DeckController.cs]: yep

**Review by KarolisPetrovskis — APPROVED** [review]

**Pull request merged**